pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -26,38 +35,46 @@ type state interface {
 
 // Returns the state towards which the protocol will transition to if the msgType is processed.
 func stateFromMsgType(msgType string) (state, error) {
-	var s state
 	switch msgType {
 	case connectionInvite:
-		s = &invited{}
+		return &invited{}, nil
 	case connectionRequest:
-		s = &requested{}
+		return &requested{}, nil
 	case connectionResponse:
-		s = &responded{}
+		return &responded{}, nil
 	case connectionAck:
-		s = &completed{}
+		return &completed{}, nil
 	default:
 		return nil, fmt.Errorf("unrecognized msgType: %s", msgType)
 	}
-	return s, nil
 }
 
 // Returns the state representing the name.
 func stateFromName(name string) (state, error) {
-	all := []state{&null{}, &invited{}, &requested{}, &responded{}, &completed{}}
-	for _, s := range all {
-		if s.Name() == name {
-			return s, nil
-		}
+	switch name {
+	// TODO: need clarification: noOp state was missing, was it a bug or feature?
```

Comment:
@llorllale

Reply:
It wasn't a bug - there's no use case for it inside this function